aes/src/armv8.rs

```diff
@@ -17,20 +17,55 @@ mod expand;
 #[cfg(test)]
 mod test_expand;
 
-use self::{
-    encdec::{decrypt1, decrypt8, encrypt1, encrypt8},
-    expand::{expand_key, inv_expanded_keys},
-};
-use crate::{Block, Block8};
 use cipher::{
-    consts::{U16, U24, U32, U8},
-    inout::InOut,
-    AlgorithmName, BlockBackend, BlockCipher, BlockCipherDecrypt, BlockCipherEncrypt, BlockClosure,
-    BlockSizeUser, Key, KeyInit, KeySizeUser, ParBlocksSizeUser,
+    consts::{self, U16, U24, U32},
+    AlgorithmName, BlockCipher, BlockCipherDecrypt, BlockCipherEncrypt, BlockClosure,
+    BlockSizeUser, Key, KeyInit, KeySizeUser,
 };
-use core::arch::aarch64::*;
 use core::fmt;
 
+impl_backends!(
+    enc_name = Aes128BackEnc,
+    dec_name = Aes128BackDec,
+    key_size = consts::U16,
+    keys_ty = expand::Aes128RoundKeys,
+    par_size = consts::U15,
```

Comment:
ARMv8 NEON has 32 SIMD registers, so techincally we can process 21, 19, and 17 blocks in parallel for AES-128, 192, and 256 respectively while keeping round keys in registers. But since the code forces inlining, it will balloon binary size, so additional benchmarks are needed.